Sources/Lang.php

```diff
@@ -505,6 +499,60 @@ public static function get(bool $use_cache = true): array
 		return Utils::$context['languages'];
 	}
 
+	/**
+	 * High level method that retrieves language strings, inserts any arguments
+	 * into them, and then returns the resulting finalized string.
+	 *
+	 * If no arguments are supplied in $args, this method simply returns the
+	 * requested string.
+	 *
+	 * If arguments are supplied in $args, this method will insert them into the
+	 * requested string. The method can handle argument substitution for both
+	 * MessageFormat and sprintf format strings.
+	 *
+	 * @param string|array $txt_key The key of the Lang::$txt array that
+	 *    contains the desired string. If this is an array, each item of the
+	 *    array will be used as a sub-key to drill down into deeper levels of
+	 *    the overall array.
+	 * @param array $args Arguments to substitute into the Lang::$txt string.
+	 * @param array $var Name of the array to search in. Default: 'txt'.
+	 *    Other possible values are 'helptxt', 'editortxt', and 'tztxt'.
+	 * @return string The string to display to the user.
+	 */
+	public static function getTxt(string|array $txt_key, array $args = [], string $var = 'txt'): string
+	{
+		// Validate $var.
+		if (!in_array($var, ['txt', 'tztxt', 'editortxt', 'helptxt', 'txtBirthdayEmails'])) {
+			throw new \ValueError();
+		}
+
+		// Don't waste time when getting a simple string.
+		if ($args === [] && is_string($txt_key)) {
+			return self::${$var}[$txt_key] ?? '';
+		}
+
+		// Trying to get something more complex...
+		$txt_key = (array) $txt_key;
+
+		$target = &self::${$var};
+
+		// Drill down to the specified key.
+		foreach ($txt_key as $key) {
+			if (isset($target[$key])) {
+				$target = &$target[$key];
+			} else {
+				// Can't do anything with a bad key.
+				return '';
+			}
+		}
+
+		if (empty($args)) {
+			return $target;
+		}
+
+		return self::formatText($target, $args);
```

Comment:
Same possible issue could occur here.  Take for example if we called `Lang::getTxt(['days'], []);`  We would have `$target` essentially pointing to `$txt['days']`.  Logic would not be happy here.

Reply:
Good catch. I hadn't thought of that scenario. If $target is an array at this point, we should throw a ValueError.